Source/cmake/WebKitCompilerFlags.cmake

```diff
@@ -458,6 +458,28 @@ int main() {
         unset(CMAKE_REQUIRED_LIBRARIES)
     endif ()
 
+    # In 32-bit architectures, data types larger than 8 bytes require libatomic.
+    if (ATOMICS_ARE_BUILTIN)
+        set(ATOMICS_ARE_BUILTIN FALSE)
+        set(ATOMIC_TEST_SOURCE "
+#include <atomic>
+#include <optional>
+
+int main(int argc, char* argv[])
+{
+    std::atomic<std::optional<double>> d;
```

Comment:
We already have an `ATOMICS_REQUIRE_LIBATOMIC` variable defined in CMake. We should change the check that calculates it if needed, instead of adding a new one.

Reply:
I prefer not to mix up this program with the existing one, because it will make the code more less clearer. I think it's better to have different programs for different scenarios.

Reply:
Thing is, the previous program should have _already_ worked to detect whether libatomic is needed or not. So it needs fixing.

Reply:
No, we don't want to do that. It may happen again in the future that someone uses an atomic data type larger than 8-bytes and the build will break again in 32-bit. The check that determines whether to link with libatomic has to be fixed.

Reply:
I merged this additonal check into the existing program.